custom_components/mikrotik_router/mikrotik_controller.py

```diff
@@ -634,3 +665,229 @@ def get_queue(self):
             upload_burst_time, download_burst_time = self.data["queue"][uid]["burst-time"].split('/')
             self.data["queue"][uid]["upload-burst-time"] = upload_burst_time
             self.data["queue"][uid]["download-burst-time"] = download_burst_time
+
+    # ---------------------------
+    #   get_dhcp
+    # ---------------------------
+    def get_dhcp(self):
+        """Get DHCP data from Mikrotik"""
+
+        self.data["dhcp-server"] = parse_api(
+            data=self.data["dhcp-server"],
+            source=self.api.path("/ip/dhcp-server"),
+            key="name",
+            vals=[
+                {"name": "name"},
+                {"name": "interface", "default": ""},
+            ]
+        )
+
+        # Build list of local DHCP networks
+        dhcp_networks = parse_api(
```

Comment:
lets create and use new api_parse in new async function get_dhcp_networks for arp self.data["dhcp_networks"] so it can be reused elsewhere.

Reply:
available as self.data["dhcp-network"]